pkg/policy/repository.go

```diff
@@ -451,6 +453,10 @@ func (p *Repository) ReplaceByResourceLocked(rules api.Rules, resource ipcachety
 
 func (p *Repository) insert(r *rule) {
 	p.rules[r.key] = r
+	if _, ok := p.rulesByNamespace[r.key.resource.Namespace()]; !ok {
```

Comment:
Is it possible for resource to be `nil`?

Reply:
> Is it possible for resource to be nil?

AFAIK it can only be an empty string. In that case, it assumes cluster-wide in `func (r ResourceID) Namespace()`

Reply:
Also, I don't see how this would work with empty resource and potentially conflicting keyIDs :thinking: 
```
	if len(rid) > 0 {
		if p.rulesByResource[rid] == nil {
			p.rulesByResource[rid] = map[ruleKey]*rule{}
		}
		p.rulesByResource[rid][r.key] = r
	}
```

Reply:
So I guess, if a rule doesn't have resource ("len(r.key.resource) == 0) we just drop the rule. I can do the same for my thing.

Reply:
okay, I dig into it, so how it works as far as I can see:
- either here: https://github.com/cilium/cilium/blob/a7ae12c7e11f4801b058e171e8ccfc072d19b524/pkg/policy/repository.go#L426 where we always have resource != ""
- or here: https://github.com/cilium/cilium/blob/a7ae12c7e11f4801b058e171e8ccfc072d19b524/pkg/policy/repository.go#L410
where we don't have resource (so default == empty string ""), but we have unique `nextID`

The second case is only for network policies added via local API, so it seems acceptable to add them as cluster-wide network policies (aka with "" namespace)

Reply:
Great, thanks for checking. Just wanted to make sure we didn't have any rude surprises.